ckanext/canada/tables/grants.yaml

```diff
@@ -900,8 +979,44 @@ resources:
           errors := errors || required_error(NEW.description_en, 'description_en');
           errors := errors || required_error(NEW.description_fr, 'description_fr');
         END IF;
+
+        IF NEW.agreement_start_date >= '2025-11-01'::date THEN
+          errors := errors || conditional_required_error(NEW.recipient_type, 'recipient_type');
+
+          errors := errors || conditional_required_error(NEW.prog_name_en, 'prog_name_en');
+          errors := errors || conditional_required_error(NEW.prog_name_fr, 'prog_name_fr');
+
+          errors := errors || conditional_required_error(NEW.prog_purpose_en, 'prog_purpose_en');
+          errors := errors || conditional_required_error(NEW.prog_purpose_fr, 'prog_purpose_fr');
+
+          errors := errors || conditional_required_error(NEW.agreement_title_en, 'agreement_title_en');
+          errors := errors || conditional_required_error(NEW.agreement_title_fr, 'agreement_title_fr');
+
+          errors := errors || conditional_required_error(NEW.agreement_end_date, 'agreement_end_date');
+          IF NEW.agreement_start_date::timestamp > NEW.agreement_end_date::timestamp THEN
+            errors := errors || ARRAY[['agreement_start_date', {start_end_date_error}]];
+          END IF;
+
+          errors := errors || conditional_required_error(NEW.expected_results_en, 'expected_results_en');
+          errors := errors || conditional_required_error(NEW.expected_results_fr, 'expected_results_fr');
+
+          NEW.recipient_postal_code := trim_lead_trailing(NEW.recipient_postal_code);
+          NEW.recipient_postal_code := UPPER(NEW.recipient_postal_code);
+          NEW.recipient_postal_code := REGEXP_REPLACE(NEW.recipient_postal_code, '\s+', ' ', 'g');
+          IF NEW.recipient_postal_code IS NOT NULL AND NEW.recipient_postal_code <> '' AND NOT NEW.recipient_postal_code ~ '^[A-Z]\d[A-Z]\s\d[A-Z]\d$' THEN
```

Comment:
For the API/form this accepts any whitespace in between, we want only a normal spaces here IIUC

Reply:
Oh I see the replace above now